src/components/InventoryGroups/Modals/AddSystemsToGroupModal.js

```diff
@@ -92,14 +111,75 @@ const AddSystemsToGroupModal = ({
     dispatch(clearFilters());
   };
 
+  const edgeParityEnabled =
+    edgeParityIsAllowed &&
+    useFeatureFlag('edgeParity.inventory-list') &&
+    useFeatureFlag('edgeParity.inventory-groups-enabled');
+
+  const [selectedImmutableDevices, setSelectedImmutableDevices] = useState([]);
+  const selectedImmutableKeys = selectedImmutableDevices.map(
+    (immutableDevice) => immutableDevice.id
+  );
+
+  // overallSelectedKeys is the list of the conventional and immutable systems ids
+  const overallSelectedKeys = [...selected.keys(), ...selectedImmutableKeys];
+  // noneSelected a boolean showing that no system is selected
+  const noneSelected = overallSelectedKeys.length === 0;
+
+  const immutableDevicesAlreadyHasGroup = selectedImmutableDevices.filter(
+    (immutableDevice) => immutableDevice.deviceGroups?.length > 0
+  );
+  // show warning also when immutable systems had groups
+  showWarning = showWarning || immutableDevicesAlreadyHasGroup.length > 0;
+
+  const [hasImmutableSystems, setHasImmutableSystems] = useState(false);
+
+  useEffect(() => {
+    if (edgeParityEnabled) {
+      (async () => {
+        const result = await axios.get(
```

Comment:
We should definitely keep this data somewhere on a higher level and share it across components where needed. Otherwise, this same request to check if there is a system is made in multiple places on a single page which is not optimal.

Reply:
What you mean "keep this data" ?

Reply:
created a function inventoryHasEdgeSystems at src/Utilities/edge.js